### ISSUE TITLE
binary classification get all 0s with GRU

### ISSUE THREAD

**Issue by jason887** [fchollet/keras#3880]
I am studying Chinese stock market with  binary classification with GRU, the traing samples is balance and the output is fine with class 1s and 0s, however, when i want to predict the real, almost all  return is all 0s.
 It is very strange the i have one mode can predict 2 classes with oneday, the predict output is similar to training data, which is fine.
when this mode is trained with more epoches, I try to use it to predict the same day data, it's predict data is all like 0.2-0.3, which means I get all 0s?
I prepare all real data with same code so it is hard to explain one day is fine and the others are wrong
Any idea?thanks

**Comment by ddofer** [comment]
Are you using softmax as the output layer? 
If it's binary and just 1 output node, it'll normalize all the outputs to the same value. USe 'Sigmoid'. 

Alternatively - what's the class distribution like?
